### PR TITLE
serde YAML/JSON input

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -73,6 +73,20 @@ impl Default for Target {
     }
 }
 
+#[allow(unused)]
```
Comment (Emilgardis) [inline src/util.rs:76]: Why allow unused? Should this not be defined in `main.rs` maybe?

File: src/lib.rs [hunk 2]
```diff
@@ -532,7 +532,14 @@ pub enum SvdError {
...
 
-    let device = svd_parser::parse(xml)?;
+    let mut parser_config = svd_parser::Config::default();
+    parser_config.validate_level = if config.strict {
+        svd::ValidateLevel::Strict
+    } else {
+        svd::ValidateLevel::Weak
+    };
+
+    let device = svd_parser::parse_with_config(xml, &parser_config)?;
```
Comment (Emilgardis) [inline src/lib.rs:542]: oops :D

this should probably be mentioned in the changelog. 

> Correctly use config when using `svd2rust::generate` as a library

Comment (Emilgardis) [reply src/lib.rs:542]: also, why set it as default and then override it directly afterwards?

**Comment by burrbull** [comment]
Should be fixed

File: src/util.rs [hunk 3]
```diff
@@ -14,6 +14,26 @@ pub const BITS_PER_BYTE: u32 = 8;
 /// that are not valid in Rust ident
 const BLACKLIST_CHARS: &[char] = &['(', ')', '[', ']', '/', ' ', '-'];
 
+pub(crate) fn load_from(input: &str, config: &crate::util::Config) -> Result<Device> {
```
Comment (Emilgardis) [inline src/util.rs:17]: I dont understand why this is living in util.rs instead if lib.rs

Comment (burrbull) [reply src/util.rs:17]: I've tried to move `load_from` to `lib.rs` but get several errors looking like:

```
    = note: expected reference `&util::Config`
               found reference `&svd2rust::Config`
For more information about this error, try `rustc --explain E0308`.
```

Could you try it yourself?

Comment (Emilgardis) [reply src/util.rs]: done, did some other changes as well.

the issue was that the util mod in main.rs and the util mod in lib.rs were separate things, and that means that the types were not equal.

**Comment by burrbull** [comment]
Rebased

**Comment by therealprof** [comment]
How would one use this? Are there parsable examples we could/should include?

**Comment by burrbull** [comment]
Espressif test was disabled. See https://github.com/esp-rs/xtensa-lx/issues/14

**Comment by burrbull** [comment]
For testing with CI. New `svdtools` release needed first.

**Review by adamgreig — APPROVED** [review]
LGTM, nice!

bors merge

**Pull request merged**